src/components/product-icon/index.tsx

```diff
@@ -4,40 +4,79 @@
  */
 
 import { IconBoundary16 } from '@hashicorp/flight-icons/svg-react/boundary-16'
+import { IconBoundary24 } from '@hashicorp/flight-icons/svg-react/boundary-24'
 import { IconConsul16 } from '@hashicorp/flight-icons/svg-react/consul-16'
+import { IconConsul24 } from '@hashicorp/flight-icons/svg-react/consul-24'
 import { IconHcp16 } from '@hashicorp/flight-icons/svg-react/hcp-16'
+import { IconHcp24 } from '@hashicorp/flight-icons/svg-react/hcp-24'
 import { IconNomad16 } from '@hashicorp/flight-icons/svg-react/nomad-16'
+import { IconNomad24 } from '@hashicorp/flight-icons/svg-react/nomad-24'
 import { IconPacker16 } from '@hashicorp/flight-icons/svg-react/packer-16'
+import { IconPacker24 } from '@hashicorp/flight-icons/svg-react/packer-24'
 import { IconTerraform16 } from '@hashicorp/flight-icons/svg-react/terraform-16'
+import { IconTerraform24 } from '@hashicorp/flight-icons/svg-react/terraform-24'
 import { IconVagrant16 } from '@hashicorp/flight-icons/svg-react/vagrant-16'
+import { IconVagrant24 } from '@hashicorp/flight-icons/svg-react/vagrant-24'
 import { IconVault16 } from '@hashicorp/flight-icons/svg-react/vault-16'
+import { IconVault24 } from '@hashicorp/flight-icons/svg-react/vault-24'
 import { IconWaypoint16 } from '@hashicorp/flight-icons/svg-react/waypoint-16'
+import { IconWaypoint24 } from '@hashicorp/flight-icons/svg-react/waypoint-24'
 import { ProductIconProps } from './types'
 
 const productSlugsToIcons = {
```

Comment:
💭 I wonder if it would be better to make a 'small' and 'large' separate versions of this component so we aren't importing double the icons with a component that is so widely used throughout the app.

Reply:
As far as I can tell from the bundle analysis comment only the home page's size is increasing, so I don't think other references to this component are significantly affected.

Reply:
In running the bundle analyzer, looks like the component increased in size by about 3kb.  So probably not a huge deal. I remember having issues with this before, but maybe webpacks code splitting is just getting better 😆  but just from a component interface perspective, I think it would be more composable if we created a 'large' version of this component instead of having the conditional. Especially since this is the first instance of needing the large logos. 

_Upstream_
<img width="1329" alt="Screenshot 2023-03-27 at 10 31 52 AM" src="https://user-images.githubusercontent.com/36613477/228023052-d00a9909-1b1f-4129-825d-886554b0718c.png">

_This branch_
<img width="1356" alt="Screenshot 2023-03-27 at 10 41 18 AM" src="https://user-images.githubusercontent.com/36613477/228023070-d6505755-813b-4423-9f75-e7e37a3beacb.png">

Reply:
I think since `size` isn't an unprecedented component prop in our library, the change is totally backwards compatible, and the overall bundle size isn't affected, I'm going to leave this in here. There are more PRs for the home page coming up after this one, so there'll be time to continue considering the prop! And we can also test and see if this causes any major disruptions.